src/net_processing.cpp

```diff
@@ -3725,6 +3743,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
         peer->m_getaddr_recvd = true;
 
+        SetupAddressRelay(pfrom);
```

Comment:
Perhaps move this above the `m_getaddr_recvd` logic (to avoid reading the unused `m_getaddr_recvd` value for non-addr-relay peers).

Reply:
I don't follow- there is an earlier clause that returns early if `!IsInboundConn`, so `SetupAddressRelay` should never return false. So I don't think the order matters here?

Reply:
ah gotcha, ok! moved